modules/dpm/src/dpm_cascade_detector.cpp

```diff
@@ -0,0 +1,159 @@
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+
+#include "precomp.hpp"
+#include "dpm_cascade.hpp"
+
+namespace cv
+{
+namespace dpm
+{
+
+class DPMDetectorImpl : public DPMDetector
+{
+public:
+
+    DPMDetectorImpl( const std::vector<std::string>& filenames, const std::vector<std::string>& classNames=std::vector<std::string>() );
+    ~DPMDetectorImpl();
+
+    bool isEmpty() const;
+
+    void detect(cv::Mat &image, CV_OUT std::vector<ObjectDetection>& objects, float overlapThreshold=0.5f);
+
+    const std::vector<std::string>& getClassNames() const;
+    size_t getClassCount() const;
+    std::string extractModelName( const std::string& filename );
+
+private:
+    std::vector<DPMCascade*> detectors;
+    std::vector<std::string> classNames;
+};
+
+cv::Ptr<DPMDetector> DPMDetector::create(std::vector<std::string> const &filenames,
+                                     std::vector<std::string> const &classNames)
+{
+    return cv::makePtr<DPMDetectorImpl>(filenames, classNames);
+}
+
+DPMDetectorImpl::ObjectDetection::ObjectDetection() : score(0.f), classID(-1) {}
+
+DPMDetectorImpl::ObjectDetection::ObjectDetection( const Rect& _rect, float _score, int _classID ) :
+    rect(_rect), score(_score), classID(_classID) {}
+
+
+DPMDetectorImpl::DPMDetectorImpl( const std::vector<std::string>& filenames, const std::vector<std::string>& _classNames )
+{
+    for( size_t i = 0; i < filenames.size(); i++ )
+    {
+        const std::string filename = filenames[i];
+        if( filename.length() < 5 || filename.substr(filename.length()-4, 4) != ".xml" )
+            continue;
+
+        DPMCascade* detector = new DPMCascade();
```

Comment:
In general whereever you say "new", there should be a corresponding delete. Now you have a tiny memory leak. 

Using here Ptr<DPMCascade> would solve it simply. Whenever you have raw pointers, make an effort to replace them with Ptr<...>